### ISSUE TITLE
Tugboat content builds fail with error "husky: Permission denied"

### ISSUE THREAD

**Issue by acrollet** [department-of-veterans-affairs/va.gov-cms#5788]
**Describe the defect**
Content builds with the current master of `content-build` fail with the following error:

```
/bin/sh: 1: husky: Permission denied
error Command failed with exit code 126.
```

**To Reproduce**
1. Go to '/admin/content/deploy'
2. Click on 'Release Content'
4. See error in logs after build fails

**Expected behavior**
Build completes successfully

**Additional context**
It appears that this change was introduced here: https://github.com/department-of-veterans-affairs/content-build/blame/763e2933268ca4e591e0ae76e1c67326a2ee2e29/package.json#L57

## Suggested resolution
Modify the command builder to include `web/node_modules/.bin` in the `PATH` environment variable.

**Comment by indytechcook** [comment]
WIth the path suggestions, the error is `permission denied`, not `command not found` so I think it's something else.  But I could be wrong.

**Comment by indytechcook** [comment]
I created a clone of master in pull and demo setups.  Both setups had no issues building front ends on different branches

https://tugboat.vfs.va.gov/60e70fb54fa323da33d96571

![image.png](https://images.zenhubusercontent.com/543e76bb7c6ccc6277a54276/b9357a1f-ec5f-423a-a34e-4c725c3259b7)

This is the same branch others have issues with.  It might be "fixed" now.

**Comment by indytechcook** [comment]
I think the file which is losing write permission is `web/.husky/_/husky.sh`

**Comment by acrollet** [comment]
The original reporter is not still experiencing the issue and we cannot reproduce with the latest code, closing this out.

**Comment by ElijahLynn** [comment]
@mpelzsherman came across this again today, and we aren't sure why it isn't happening all the time though. However, we did realize what is happening.

There is a `husky install` in the `prepare` command/script of package.json that gets implicitly invoked by a `yarn install` (reproducible by running just this). I was able to run an `strace --follow-forks yarn install` from the ./web directory and found: 

```
[pid  3182] clone(strace: Process 3183 attached
child_stack=NULL, flags=CLONE_CHILD_CLEARTID|CLONE_CHILD_SETTID|SIGCHLD, child_tidptr=0x7fbfe23e2850) = 3183
[pid  3182] wait4(-1,  <unfinished ...>
[pid  3183] execve("/var/lib/tugboat/docroot/vendor/va-gov/content-build/node_modules/.bin/husky", ["husky", "install"], 0x56228b666518 /* 346 vars */) = -1 EACCES (Permission denied)
```

Then:

```
root@php:/var/lib/tugboat/web# ls -al /var/lib/tugboat/docroot/vendor/va-gov/content-build/node_modules/.bin/husky
lrwxrwxrwx 1 www-data www-data 19 Aug  2 21:15 /var/lib/tugboat/docroot/vendor/va-gov/content-build/node_modules/.bin/husky -> ../husky/lib/bin.js
```
```
root@php:/var/lib/tugboat/web# ls -al /var/lib/tugboat/docroot/vendor/va-gov/content-build/node_modules/husky/lib/bin.js 
-rw-rw-r-- 1 www-data www-data 1030 Aug  2 21:15 /var/lib/tugboat/docroot/vendor/va-gov/content-build/node_modules/husky/lib/bin.js
```

Then I ran a `chmod u+x` on bin.js and the `yarn install` (which runs `husky install` via the `yarn prepare` command) worked:

```
root@php:/var/lib/tugboat/web# yarn install
yarn install v1.21.1
[1/4] Resolving packages...
warning Resolution field "lodash@4.17.21" is incompatible with requested version "lodash@^2.4.1"
success Already up-to-date.
$ husky install
/bin/sh: 1: husky: Permission denied
error Command failed with exit code 126.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
root@php:/var/lib/tugboat/web# chmod 777 /var/lib/tugboat/docroot/vendor/va-gov/content-build/node_modules/husky/lib/bin.js 
root@php:/var/lib/tugboat/web# yarn install
yarn install v1.21.1
[1/4] Resolving packages...
warning Resolution field "lodash@4.17.21" is incompatible with requested version "lodash@^2.4.1"
success Already up-to-date.
$ husky install
husky - Git hooks installed
Done in 0.68s.
```

I opened an upstream issue for possibly improving the documentation on the install of `husky`. 

And lastly, we need to automatically make it executable in either our CI scripts or the upstream content-repo scripts.

**Comment by indytechcook** [comment]
This isn't an issue for all build.  Not even all `branch` builds.  

At one time I thought I had found a pattern where if you do a `branch` build before a `default` build it broke, but even that wasn't the case all the time.   Our PR environment are never able to reproduce the issue which made me think the blanket `chown/chmod` we do after the content build in the `online` step was fixing the issue but I tried this on a clone and it still wasn't able to reproduce the issue consistently.
